internal/pkg/util/fs/helper.go

```diff
@@ -134,34 +134,53 @@ func RootDir(path string) string {
 	return p
 }
 
-// EvalRelative evaluates symlinks in path relative to root path. This
-// function doesn't return error but always returns an evaluated path.
+// EvalRelative evaluates symlinks in path relative to root path, it returns
+// a path as if it was evaluated from chroot. This function always returns
+// an absolute path.
 func EvalRelative(path string, root string) string {
-	splitted := strings.Split(filepath.Clean(path), string(os.PathSeparator))
-	dest := string(os.PathSeparator)
-
-	for i := 1; i < len(splitted); i++ {
-		s := splitted[i : i+1][0]
-		dest = filepath.Join(dest, s)
-
-		if s != "" {
-			rootDestPath := filepath.Join(root, dest)
-			for {
-				target, err := filepath.EvalSymlinks(rootDestPath)
-				if err != nil {
-					break
-				}
-				if !strings.HasPrefix(target, root) {
-					rootDestPath = filepath.Join(root, target)
-					continue
+	cleanRoot := filepath.Clean(root)
+	cleanPath := filepath.Clean(path)
+
+	if cleanPath == "" {
+		return cleanRoot
+	}
+
+	sep := string(os.PathSeparator)
+	splitted := strings.Split(cleanPath, sep)
+
+	i := 1
```

Comment:
```
i := 0
if filepath.IsAbs(cleanPath) {
    // path is absolute, skip the first empty element
    // that resulted from calling strings.Split above.
    i = 1
}
```

note that the index itself is not really important, so this is more or less equivalent to doing something like:

```
dest := cleanRoot

components := strings.Split(cleanPath, string(os.PathSeparator))
if filepath.IsAbs(cleanPath) {
    components = components[1:]
}

for _, component := range components {
}
```